readme.md

```diff
@@ -27,6 +27,32 @@ is.number(6);
 //=> true
 ```
 
+When using `is` together with TypeScript, [type guards](http://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types) are being used to infer the correct type inside if-else statements.
```

Comment:
In VSCode, type inference also works in JavaScript. Don't think we should document this as it's IDE specific.